pootle/static/js/shared/components/FormSelectInput.js

```diff
@@ -41,19 +44,23 @@ const FormSelectInput = React.createClass({
   /* Handlers */
 
   handleChange(value) {
-    this.props.handleChange(this.props.name, value);
+    const newValue = this.props.multiple ? value.split(',') : value;
+    this.props.handleChange(this.props.name, newValue);
   },
 
 
   /* Layout */
 
   render() {
+    const { value } = this.props;
+    /* FIXME: react-select#25 prevents using non-string values */
+    const selectValue = this.props.multiple ? value : value.toString();
     return (
       <Select
         onChange={this.handleChange}
         {...this.props}
-        /* FIXME: react-select#25 prevents using non-string values */
-        value={this.props.value.toString()}
+        multi={this.props.multiple}
```

Comment:
Being a mirror of react-select's API is definitely not a goal of`FormElement` + `FormSelectInput` components, but rather provide a thin abstraction with sane defaults (I know this for a fact because I wrote them). In that context, [`multiple` is a natural and well-known choice](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/select#attr-multiple) to denote that a select element accepts multiple values.

The only code which cares and has to deal directly with react-select is `FormSelectInput`, it uses `multi={<bool>}` and if that's hard to understand, I'm definitely missing the point.

Having said that, I have other simplifications for `FormElement` in mind which will make it easier to reason about the rendering of form controls — after all this is one of the initial React components I wrote around 2 years ago, and best practices, common patterns and experience have evolved since then.